src/ntfs/write.rs

```diff
@@ -179,6 +175,48 @@ pub fn write_fares_v1(
     Ok(())
 }
 
+fn write_fares_v1_from_v2(
+    base_path: &path::Path,
+    tickets: &CollectionWithId<Ticket>,
+    ticket_uses: &CollectionWithId<TicketUse>,
+    ticket_prices: &Collection<TicketPrice>,
+    ticket_use_perimeters: &Collection<TicketUsePerimeter>,
+    ticket_use_restrictions: &Collection<TicketUseRestriction>,
+) -> Result<()> {
+    info!("xxxxxxx write_fares_v1_from_v2");
+    Ok(())
+}
+
+pub fn write_fares_v1(
+    base_path: &path::Path,
+    prices_v1: &Collection<PriceV1>,
+    od_fares_v1: &Collection<ODFareV1>,
+    fares_v1: &Collection<FareV1>,
+    tickets: &CollectionWithId<Ticket>,
+    ticket_uses: &CollectionWithId<TicketUse>,
+    ticket_prices: &Collection<TicketPrice>,
+    ticket_use_perimeters: &Collection<TicketUsePerimeter>,
+    ticket_use_restrictions: &Collection<TicketUseRestriction>,
+) -> Result<()> {
+    if !tickets.is_empty()
+        && !ticket_uses.is_empty()
+        && !ticket_prices.is_empty()
+        && !ticket_use_perimeters.is_empty()
+    {
+        return write_fares_v1_from_v2(
+            base_path,
+            tickets,
+            ticket_uses,
+            ticket_prices,
+            ticket_use_perimeters,
+            ticket_use_restrictions,
+        );
+    } else if !prices_v1.is_empty() && !od_fares_v1.is_empty() {
```

Comment:
Similarly, you could write a `has_fares_v1()` function. Even if it's only for factorizing much less lines and it would be used only once, I still find it useful because only the name of the function documents what you're doing.